contracts/contracts/strategies/AerodromeEthStrategy.sol

```diff
@@ -0,0 +1,592 @@
+         * created is no longer valid.
+         */
+
+        uint256 poolWETHBalance = wethCoinIndex == 0
+            ? lpTokenAddress.reserve0()
+            : lpTokenAddress.reserve1();
+
+        /* K is multiplied by 1e36 which is used for higher precision calculation of required
+         * pool LP tokens. Without it the end value can have rounding errors up to precision of
+         * 10 digits. This way we move the decimal point by 36 places when doing the calculation
+         * and again by 36 places when we are done with it.
+         */
+        uint256 k = (1e36 * IERC20(address(lpTokenAddress)).totalSupply()) /
+            poolWETHBalance;
+        // prettier-ignore
+        // slither-disable-next-line divide-before-multiply
+        uint256 diff = (_wethAmount + 1) * k;
+        lpToBurn = diff / 1e36;
+    }
+
+    /**
+     * @notice Remove all ETH and OETH from the Aero pool, burn the OETH,
+     * convert the ETH to WETH and transfer to the Vault contract.
+     */
+    function withdrawAll() external override onlyVaultOrGovernor nonReentrant {
+        uint256 gaugeTokens = aeroGaugeAddress.balanceOf(address(this));
+        _lpWithdraw(gaugeTokens);
+
+        // Remove liquidity
+        // slither-disable-next-line unused-return
+        aeroRouterAddress.removeLiquidity(
+            address(weth),
+            address(oeth),
+            true,
+            gaugeTokens,
+            0,
+            0,
+            address(this),
+            block.timestamp
+        );
+
+        // Burn all OETH
+        uint256 oethToBurn = oeth.balanceOf(address(this));
+        IVault(vaultAddress).burnForStrategy(oethToBurn);
+
+        // Get the strategy contract's wrapped ether balance and transfer to the vault.
+        // This includes all that was removed from the AERO pool and
+        // any ether that was sitting in the strategy contract before the removal.
+        uint256 wethBalance = weth.balanceOf(address(this));
+        // Convert all the strategy contract's ether to WETH
+        require(
+            weth.transfer(vaultAddress, wethBalance),
+            "Transfer of WETH not successful"
+        );
+
+        emit Withdrawal(address(weth), address(lpTokenAddress), wethBalance);
+        emit Withdrawal(address(oeth), address(lpTokenAddress), oethToBurn);
+    }
+
+    // Pool Peg keeping
+
+    /**
+     * @notice Mints and swaps the given oeth token amount for weth to maintain pool balance.
+     * @param _amountIn Amount of tokens to swap for.
+     * @param _minAmountOut Minimum amount of expected output amount (slippage adjusted)
+     * @param _tokenIn Address of the input token.
+     * @param _recipient Address of the recipient of weth tokens if we swap oeth for weth.
+     */
+    function swapAndRebalancePool(
+        uint256 _amountIn,
+        uint256 _minAmountOut,
+        address _tokenIn,
+        address _recipient
```

Comment:
The strategist should not be able to steal tokens from the strategy / vault. We should probably remove the `_recipient` option here.

Reply:
If we remove the _recipient, where should the swapped `weth` sent? Will it be held in the AMO contract?